x/ibchooks/ibc_hooks.go

```diff
@@ -0,0 +1,87 @@
+package ibchooks
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+	clienttypes "github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v6/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v6/modules/core/exported"
+	ibckeeper "github.com/cosmos/ibc-go/v6/modules/core/keeper"
+
+	"github.com/provenance-io/provenance/x/ibchooks/keeper"
+	"github.com/provenance-io/provenance/x/ibchooks/types"
+)
+
+type IbcHooks struct {
+	cdc                     codec.BinaryCodec
+	ibcKeeper               *ibckeeper.Keeper
+	ibcHooksKeeper          *keeper.Keeper
+	wasmHooks               *WasmHooks
+	markerHooks             *MarkerHooks
+	SendPacketPreProcessors []types.PreSendPacketDataProcessingFn
+}
+
+func NewIbcHooks(cdc codec.BinaryCodec, ibcHooksKeeper *keeper.Keeper, ibcKeeper *ibckeeper.Keeper, wasmHooks *WasmHooks, markerHooks *MarkerHooks, preSendPacketDataProcessingFns []types.PreSendPacketDataProcessingFn) IbcHooks {
+	return IbcHooks{
+		cdc:                     cdc,
+		ibcKeeper:               ibcKeeper,
+		ibcHooksKeeper:          ibcHooksKeeper,
+		wasmHooks:               wasmHooks,
+		markerHooks:             markerHooks,
+		SendPacketPreProcessors: preSendPacketDataProcessingFns,
+	}
+}
+
+// ProperlyConfigured returns false if either wasm or marker hooks are configured properly
```

Comment:
Minor typo. I think you mean this:

ProperlyConfigured returns false if either wasm or marker hooks are configured **incorrectly**.